Validation/HGCalValidation/plugins/HGCalBHValidation.cc

```diff
@@ -122,21 +119,19 @@ void HGCalBHValidation::analyze(const edm::Event& e, const edm::EventSetup&) {
       double energy = it->energy();
```

Comment:
This whole loop could be more conveniently and less error prone rewritten as
```C++
    for (auto const& it : hitsHE) {
      double energy = it.energy();
      double time = it.time();
      unsigned int id = it.id();
      bool bh = (DetId(id).det() == DetId::HGCalHSc);
      if (bh) {
        double ring = HGCScintillatorDetId(id).ring() + 0.01;
        double phi = HGCScintillatorDetId(id).iphi() - 0.01;
        int lay = HGCScintillatorDetId(id).layer();
        double ring1 = HGCScintillatorDetId(id).zside() * ring;
        hsi2Oc_->Fill(ring1, phi, energy);
        hsimE1_->Fill(energy);
        hsimTm_->Fill(time, energy);
        hsimOc_->Fill(ring1, phi, energy);
        hsimLn_->Fill(lay, energy);
        hsi3Oc_->Fill(ring1, lay, energy);
        double ensum = map_try.count(id) != 0? map_try[id] : 0.;
        ensum += energy;
        map_try[id] = ensum;
        edm::LogVerbatim("HGCalValidation") << "HGCalBHHit[" << ++i << "] ID " << std::hex << " " << id << std::dec << " "
                                            << HGCScintillatorDetId(id) << " E " << energy << " time " << time;
      }
   }
```
And the same in the for loop right below